log4j-core/src/main/java/org/apache/logging/log4j/core/util/internal/instant/InstantPatternDynamicFormatter.java

```diff
@@ -711,23 +488,114 @@ public void formatTo(final StringBuilder buffer, final Instant instant) {
                 }
             };
         }
+
+        @Override
+        @Nullable
+        PatternFormatterFactory tryMerge(PatternFormatterFactory other, ChronoUnit thresholdPrecision) {
+            // We always merge consecutive static pattern factories
+            if (other instanceof StaticPatternFormatterFactory) {
+                final StaticPatternFormatterFactory otherStatic = (StaticPatternFormatterFactory) other;
+                return new StaticPatternFormatterFactory(this.literal + otherStatic.literal);
+            }
+            // We also merge a static pattern factory with a DTF factory
+            if (other instanceof DateTimeFormatterPatternFormatterFactory) {
+                final DateTimeFormatterPatternFormatterFactory otherDtf =
+                        (DateTimeFormatterPatternFormatterFactory) other;
+                return new DateTimeFormatterPatternFormatterFactory(
+                        this.pattern + otherDtf.pattern, otherDtf.precision);
+            }
+            return super.tryMerge(other, thresholdPrecision);
```

Comment:
One less cognitive hop:
```suggestion
            return null;
```